ui/js/search.js

```diff
@@ -101,13 +101,18 @@ require(['jquery', 'oae.core', 'underscore', 'jquery.history', 'jquery.switchtab
             $('#search-refine-type input[type="checkbox"][data-type="' + type + '"]').prop('checked', true);
         });
 
-        // Ensure only the selected tenant is chosen, if at all
-        $('#search-refine-tenant input[type="checkbox"]').removeAttr('checked');
-        if (queryData.tenant) {
-            $('#search-refine-tenant input[type="checkbox"][data-tenant="' + queryData.tenant + '"]').prop('checked', true);
+        // Hide the 'within tenant' checkbox for private tenants
+        var isPrivate = oae.api.config.getValue('oae-tenants', 'tenantprivacy', 'tenantprivate');
+        if (isPrivate) {
+            $('#search-refine-tenant').hide();
+        } else {
+            // Ensure only the selected tenant is chosen, if at all
```

Comment:
I don't think the back-end does anything fancy with assuming whether or not the UI will want to show items for the current tenant or the whole system.

The back-end is designed to just ensure the privacy/security constraints, and then what the UI wants to project to the user is more of a front-end experience detail. So it's important for the UI to say explicitly "this tenant only", "only for that tenant", etc... explicitly.